### PR TITLE
[feat]: [PLG-2503]: while enabling NG for an account via admin tool make defaultExperiecne as NG

### PR DESCRIPTION
**Pull request by kapil-harness**
## Describe your changes
while enabling NG for an account via admin tool make defaultExperiecne as NG

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/52275)

### CODE REVIEW

**Comment by kapil-harness** [comment]
trigger uts

**Review by mohitg0795 — DISMISSED** [review]
> Reviewed 3 of 3 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52275)* status: Some of the checks are too old. 3 file(s) to review. 1 of 2 approvals obtained.

File: 400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java [hunk 1]
```diff
@@ -601,9 +601,16 @@ public boolean isNextGenEnabled(String accountId) {
   }
 
   @Override
-  public Boolean updateNextGenEnabled(String accountId, boolean enabled) {
+  public Boolean updateNextGenEnabled(String accountId, boolean enabled, boolean isAdmin) {
```
Comment (neelam-harness) [inline 400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java:604]: Nit: Let's rename `enabled` to `isNgEnabled` ?

File: 400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java [hunk 2]
```diff
@@ -601,9 +601,16 @@ public boolean isNextGenEnabled(String accountId) {
...
     Account account = get(accountId);
     account.setNextGenEnabled(enabled);
+    if (isAdmin) {
+      if (enabled) {
```
Comment (neelam-harness) [inline 400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java:608]: Nit: Let's make this ternary operation ?

**Review by neelam-harness — CHANGES_REQUESTED** [review]

File: 400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java (hunk 2 above)
Comment (kapil-harness) [reply 400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java:608]: after doing that the code is not looking very readable
`account.setDefaultExperience(isAdmin ? DefaultExperience.NG : DefaultExperience.CG);`
and doing below is not possible as setDefaultExperience() setter returns void which is not allowed in a ternary ops :) 
`isNextGenEnabled ? account.setDefaultExperience(DefaultExperience.NG) : account.setDefaultExperience(DefaultExperience.CG);`

File: 400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java (hunk 1 above)
Comment (kapil-harness) [reply 400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java:604]: done

**Comment by kapil-harness** [comment]
trigger uts

**Review by neelam-harness — COMMENTED** [review]
LGTM

> Reviewed 3 of 4 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/52275#-:-NeFjCk-8jvFdjieuVWN:bb6ywmx)* status: 4 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java` line 608 at r1](https://reviewable.io/reviews/harness/harness-core/52275#gh-1325131263:-NeFizq1-8Rd21S3_pLk:b-m5appz) ([raw file](https://github.com/harness/harness-core/blob/89344d4ee016e9a1fb929797aade862d6ed60f14/400-rest/src/main/java/software/wings/service/impl/AccountServiceImpl.java#L608)):*
<details><summary><i>Previously, kapil-harness (Kapil Choudhary) wrote…</i></summary><blockquote>

after doing that the code is not looking very readable
`account.setDefaultExperience(isAdmin ? DefaultExperience.NG : DefaultExperience.CG);`
and doing below is not possible as setDefaultExperience() setter returns void which is not allowed in a ternary ops :) 
`isNextGenEnabled ? account.setDefaultExperience(DefaultExperience.NG) : account.setDefaultExperience(DefaultExperience.CG);`
</blockquote></details>

This looks fine Kapil. `account.setDefaultExperience(isAdmin ? DefaultExperience.NG : DefaultExperience.CG);`

**Comment by kapil-harness** [comment]
trigger uts

**Review by neelam-harness — APPROVED** [review]
Thank you!

**Review by mohitg0795 — APPROVED** [review]

**Comment by kapil-harness** [comment]
trigger review

**Pull request merged**